bin/log-setup.js

```diff
@@ -2,12 +2,35 @@ var fs = require('fs-extra');
 var path = require('path');
 var log4js = require('log4js');
 
-/**
- * make a log directory, just in case it isn't there.
- */
-module.exports.init = function() {
-    fs.emptyDirSync(path.join(__dirname, '../log'));
+module.exports.init = function(opts) {
+    if (opts.l) {
+        log4js.configure(opts.l);
+    } else {
+        var log4js_opts;
 
-    log4js.configure(path.join(__dirname, '../log4js.json'));
+        if (opts.d) {
+            var logfile = '/var/log/outriggerd.log';
+            if (opts.o) {
+                logfile = opts.o;
+            }
+
+            // Throw an error if we can't log to the configured log
+            // file.
+            fs.accessSync(logfile, fs.R_OK | fs.W_OK);
+
+            log4js_opts = {
+                appenders: [
+                    {type: 'file', filename: logfile}
+                ]
+            };
+        } else {
+            log4js_opts = {
+                appenders: [
+                    {type: 'console'}
+                ]
+            };
+        }
+        log4js.configure(log4js_opts);
```

Comment:
You need to add the default level of info in the config or call log4js.globalLogLevel (or something like that), otherwise log4js will default to debug level.